x-pack/plugins/cases/public/components/all_cases/utility_bar.tsx

```diff
@@ -49,13 +66,63 @@ export const CasesTableUtilityBar: FunctionComponent<Props> = React.memo(
       onActionSuccess: onRefresh,
     });
 
+    const handleNotShowAgain = () => {
+      setLocalStorageWarning(true);
+    };
+
     /**
      * At least update or delete permissions needed to show bulk actions.
      * Granular permission check for each action is performed
      * in the useBulkActions hook.
      */
     const showBulkActions = (permissions.update || permissions.delete) && selectedCases.length > 0;
 
+    const visibleCases =
+      pagination?.pageSize && totalCases > pagination.pageSize ? pagination.pageSize : totalCases;
+
+    const hasReachedMaxCases =
+      pagination.pageSize &&
+      totalCases >= MAX_DOCS_PER_PAGE &&
+      pagination.pageSize * (pagination.pageIndex + 1) >= MAX_DOCS_PER_PAGE;
```

Comment:
why `>= MAX_DOCS_PER_PAGE` instead of `> MAX_DOCS_PER_PAGE` ?

Reply:
yeah, Since it is a warning message and not any blocked action for user, I thought it would be better to show the message when user reaches the limit (=10K) to make them aware that they are at 10k cases and fetching next page action will fail.

what do you think?

Reply:
> to make them aware that they are at 10k cases and fetching next page action will fail.

This makes sense 👍 all good